### PR TITLE
Feature create history service

### PR DESCRIPTION
**Pull request by lhohenrique**

### CODE REVIEW

File: PontoFacil/PontoFacil/Converters/DateTimeToStringConverter.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace PontoFacil.Converters
+{
+    class DateTimeToStringConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, string language)
+        {
+            try
+            {
+                return value != null ? ((DateTime)value).ToString("HH:mm:ss") : "00:00:00";
```
Comment (thiago-rocha) [inline PontoFacil/PontoFacil/Converters/DateTimeToStringConverter.cs]: Create a var with `00:00:00` value, this var should be a global.

**Review by gmojunior — APPROVED** [review]

File: PontoFacil/PontoFacil/Services/ClockInService.cs [hunk 2]
```diff
@@ -33,7 +33,8 @@ public void Register(DateTime date)
         private void StartNewDay(DateTime dt)
         {
             _clockIn = new ClockIn();
-            this._clockIn.Open(dt);
+
+            this._clockIn.Open(dt, this._persistencyService.getProfile().LunchTime);
```
Comment (thiago-rocha) [inline PontoFacil/PontoFacil/Services/ClockInService.cs]: We are no longer using `this` when referring to to local attributes and methods.

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 3]
```diff
@@ -114,6 +115,17 @@ public Profile getProfile()
 
             return p != null ? p : new Profile();
         }
+
+        public List<ClockIn> GetMonthlyHistory()
+        {
+            return _repository.ClockInList.FindAll(ci => ci.Id >= new DateTime(DateTime.Now.Year, DateTime.Now.Month, 1) && ci.Id <= DateTime.Now.Date);
```
Comment (thiago-rocha) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: Could you simplify this line?

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 4]
```diff
@@ -114,6 +115,17 @@ public Profile getProfile()
...
+        }
+
+        public List<ClockIn> GetFreeHistory(DateTimeOffset startDate, DateTimeOffset endDate)
+        {
+            return _repository.ClockInList.FindAll(ci => ci.Id >= startDate.Date && ci.Id <= endDate.Date);
```
Comment (thiago-rocha) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: Could you simplify this line?

File: PontoFacil/PontoFacil/Converters/TimeSpanToStringConverter.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    class TimeSpanToStringConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, string language)
+        {
+            string time = "00:00:00";
+            try
+            {
+                if (value != null)
+                {
+                    time = (TimeSpan)value < TimeSpan.Zero ? "-" + ((TimeSpan)value).ToString(@"hh\:mm\:ss") : ((TimeSpan)value).ToString(@"hh\:mm\:ss");
```
Comment (thiago-rocha) [inline PontoFacil/PontoFacil/Converters/TimeSpanToStringConverter.cs]: Could you simplify this line?

**Review by thiago-rocha — CHANGES_REQUESTED** [review]
@lhohenrique I just add some comments on you pr, please check them.

File: PontoFacil/PontoFacil/Converters/DateTimeToStringConverter.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        public object Convert(object value, Type targetType, object parameter, string language)
+        {
+            string time = "00:00:00";
+            try
+            {
+                return value != null ? ((DateTime)value).ToString("HH:mm:ss") : time;
+            }
+            catch (Exception)
+            {
+                return time;
```
Comment (thiago-rocha) [inline PontoFacil/PontoFacil/Converters/DateTimeToStringConverter.cs]: That is some weird with this behaivor, you are using `tyr catch` as a `if-else`, will be better if you havew only one return line after the `try catch` and on the catch fire some mesage about the error.

File: PontoFacil/PontoFacil/Models/ClockIn.cs [hunk 7]
```diff
@@ -5,47 +5,112 @@ namespace PontoFacil.Models
...
+        }
+        #endregion
+
+        #region Methods
+        public void Open(DateTime dt, int lunchTime)
+        {
+            Start = Convert.ToDateTime(dt.ToString("HH:mm:ss"));
+            LunchTime = lunchTime;
+            StartLunchTime = Convert.ToDateTime("12:00:00");
+            EndLunchTime = lunchTime == 1 ? Convert.ToDateTime("13:00:00") : Convert.ToDateTime("14:00:00");
```
Comment (thiago-rocha) [inline PontoFacil/PontoFacil/Models/ClockIn.cs]: Could you extract the `Convert.ToDateTime` in a global var named `ONE_PM` and `TWO_PM` ou something like it and extract the operation to a method. The reason is make you code more readable.

File: PontoFacil/PontoFacil/Services/PersistencyService.cs [hunk 8]
```diff
@@ -114,6 +115,23 @@ public Profile getProfile()
 
             return p != null ? p : new Profile();
         }
+
+        public List<ClockIn> GetMonthlyHistory()
+        {
+            DateTime firstDayOfTheMonth = new DateTime(DateTime.Now.Year, DateTime.Now.Month, 1);
```
Comment (thiago-rocha) [inline PontoFacil/PontoFacil/Services/PersistencyService.cs]: `new DateTime(DateTime.Now.Year, DateTime.Now.Month, 1)` extract the line the a global var named `FIRST_DAY_OF_MONTH`.

**Review by thiago-rocha — CHANGES_REQUESTED** [review]
@lhohenrique I just add some comments, please check them.

**Review by thiago-rocha — APPROVED** [review]

**Pull request merged**